docs/writing_a_simple_publisher_and_subscriber.md

```diff
@@ -0,0 +1,520 @@
+4. Publishing Loop within Thread:  
+* `thread::sleep(Duration::from_millis(1000));`: Pauses the thread for 1 second (1 Hz publishing rate).  
+* `iterator = publisher_other_thread.publish_data(count).unwrap();`: Calls the `publish_data` method on the `publisher_other_thread` to publish a message with the current counter value. Increments the iterator for the next message.  
+5. Main Thread Spin:  
+* `rclrs::spin(publisher.node.clone());`: Keeps the main thread running, processing ROS 2 events and messages. Uses a cloned reference to the node to ensure it remains active even with other threads.  
+
+</details>
+</details>
+<details><summary>Having several ROS2 Rust nodes in one Package</summary>
+
+Of course, you can write for each node you want to implement its own package, and that can have it's advantages. I implore you to use some cargo tricks and add some binary targets to your `cargo.toml`. That could look like this:
+```toml
+[package]
+name = "your_package_name"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[[bin]]
+name="simple_publisher"
+path="src/simple_publisher.rs"
+[dependencies]
+rclrs = "*"
+std_msgs = "*"
+```
+
+You'll find the name of your executable and the corresponding file name under the `[[bin]]` tag. As you can see, the filename and the name you want to call your node don't have to match. Please remember to include your executable name with snake_cases. The Rust compiler will be a bit grumpy if you don't.  
+Now, by recompiling the package from the previous chapter and making it usable:  
+```shell
+cd WORKSPACE
+colcon build
+source install/setub.bash
+```
+Running the node will look like this:
+```shell
+ros2 run your_package_name simple_publisher
+```
+As you can see, you are now calling your node by the name declared in `[[bin]]` using the `name` variable.
+
+</details>
+<details><summary>Write the subscriber node</summary> 
+
+Of course, you can implement a new ROS2 Rust package for this node. You can find out how to do this in the section called 'Create a package'.
+Or you can add a new binary target to your package. To do so, just add a new `FILE.rs` to your source directory - for simplicity I'll call this file `simple_subscriber.rs` - and add a corresponding binary target to your `Cargo.toml`:
+```toml
+[[bin]]
+name="simple_subscriber"
+path="src/simple_subscriber.rs"
+```
+To construct the subscriber node, put the [following](https://gitlab.com/ros21923912/simple_ros2_node/-/blob/more_simple_nodes/src/simple_subscriber.rs?ref_type=heads) code into a `FILE.rs` - in my case its the `src/simple_subscriber.rs`:
+```rust
+use rclrs::{create_node, Context, Node, RclrsError, Subscription, QOS_PROFILE_DEFAULT};
+use std::{
+    env,
+    iter,thread,
+    sync::{Arc, Mutex},
+    time::Duration,
+};
+use std_msgs::msg::String as StringMsg;
+/// A simple ROS2 subscriber node that receives and prints "hello" messages.
```

Comment:
Because there is a section examining the code in detail, we should probably remove the more in depth descriptions from the comments. This is what the official ROS 2 docs do
https://docs.ros.org/en/foxy/Tutorials/Beginner-Client-Libraries/Writing-A-Simple-Cpp-Publisher-And-Subscriber.html#write-the-publisher-node

Notice that there is only a single comment in their actual example code
```
/* This example creates a subclass of Node and uses std::bind() to register a
* member function as a callback from the timer. */
```

Reply:
@Guelakais please address this

Reply:
I have now deleted a few comments. That should clear that up too.